dScripts/WhFans.cpp

```diff
@@ -0,0 +1,66 @@
+#include "WhFans.h"
+
+#include "RenderComponent.h"
+
+void WhFans::OnStartup(Entity* self) {
+	self->SetVar<bool>(u"alive", true);
+	self->SetVar<bool>(u"on", false);
+
+	ToggleFX(self, false);
+}
+
+void WhFans::ToggleFX(Entity* self, bool hit) {
+	std::string fanGroup = "";
+	try {
+		fanGroup = self->GetGroups()[0];
+	} catch(...) {
+		fanGroup = "";
+	}
+
+	std::vector<Entity*> fanVolumes = EntityManager::Instance()->GetEntitiesInGroup(fanGroup);
+
+	auto* renderComponent = static_cast<RenderComponent*>(self->GetComponent(COMPONENT_TYPE_RENDER));
```

Comment:
Use modern `self->GetComponent<RenderComponent*>()` instead, and check for nullptr.

Reply:
Oh man that nullptr is what was crashing the world server, Fans now work :D